src/lib.rs

```diff
@@ -32,9 +32,39 @@ use std::{
     io::{self, Write},
     panic, process,
     sync::{mpsc, Arc, RwLock},
-    thread, time,
+    thread,
 };
 
+// A panic in any thread should stop the main thread, and print the panic.
+fn setup_panic_hook() {
```

Comment:
Maybe it is a mistake, but I thought you wanted to move it to main.rs like `setup_logger`

Reply:
Why? It's used by and related to the `daemon_main` function. Do you think it should be moved to the binary? If so, why?

Reply:
Binaries embedding revaultd, may want to use daemon_main with their own panic hook. But I am ok to keep it here, It removes code on the GUI side :+1:

Reply:
I think that this is something internal to `revaultd`: we make the assumption that we can panic inside the threads we ourselves start. If someone sets the hook to something else, it might cause some weird behaviour (for instance an onchain state that doesn't get updated anymore as the bitcoind thread panic'ed and it didn't stop the main thread).

In fact, it's IMO one the most brittle things about making revaultd a library: a user of the library can rewrite this hook and by doing so would unknowingly break our assumption.